x/staking/keeper/validator_test.go

```diff
@@ -1099,3 +1055,157 @@ func TestUpdateValidatorCommission(t *testing.T) {
 		}
 	}
 }
+
+func TestRemoveTokens(t *testing.T) {
+	ctx, _, keeper := CreateTestInput(t, false, 1000)
+
+	valPubKey := PKs[0]
+	valAddr := sdk.ValAddress(valPubKey.Address().Bytes())
+
+	validator := types.Validator{
+		OperatorAddress: valAddr,
+		ConsPubKey:      valPubKey,
+		Status:          sdk.Bonded,
+		Tokens:          sdk.NewInt(100),
+		DelegatorShares: sdk.NewDec(100),
+	}
+
+	// remove tokens and test check everything
+	validator = keeper.removeTokens(ctx, validator, sdk.NewInt(10))
+	require.Equal(t, int64(90), validator.Tokens.Int64())
+
+	// update validator to from bonded -> unbonded
+	validator = keeper.UpdateStatus(ctx, validator, sdk.Unbonded)
+	require.Equal(t, sdk.Unbonded, validator.Status)
+
+	validator = keeper.removeTokens(ctx, validator, sdk.NewInt(10))
```

Comment:
I would add a require after this line to check if the validator balance got the change (like lne 1075)